### PR TITLE
maturin: 0.10.4 -> 0.10.6

### PR DESCRIPTION
**Pull request by danieldk**
###### Motivation for this change

Changelog:
https://github.com/PyO3/maturin/blob/v0.10.5/Changelog.md#0105---2021-05-21

Also add rustc to the dependencies of maturinBuildHook, otherwise maturin
cannot find the Rust compiler.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by jonringer — COMMENTED** [review]
did you want to fix darwin?

```
ld: library not found for -liconv
```

should just be `pkgs.darwin.libiconv`

**Comment by danieldk** [comment]
> did you want to fix darwin?

Ah nice, ofborg darwin builds work again?

Should be fixed now...

**Comment by jonringer** [comment]
> Ah nice, ofborg darwin builds work again?

I guess grahamc asked for some funding from the foundation to add some darwin builders.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 123867` at 11307c1d run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/c1b32a89ce5897f25efb473cff37fb3a">python38Packages.qiskit-aer</a></li>
    <li><a href="https://gist.github.com/5e8a2f61f74c9566cc85da2ab2cc1d42">python38Packages.qiskit-ibmq-provider</a></li>
    <li><a href="https://gist.github.com/8c5afb0ab1344ecbbd4f5d96f5f85484">python38Packages.qiskit-ignis</a></li>
    <li><a href="https://gist.github.com/a9f085591117de70f1ac7b0b8ad77b74">python38Packages.qiskit-terra</a></li>
    <li><a href="https://gist.github.com/d3d6c00aad0c922bb23d00f6e8ffcd40">python39Packages.qiskit-aer</a></li>
    <li><a href="https://gist.github.com/4b0b95b98568827a09e86c7e18df7e9a">python39Packages.qiskit-ibmq-provider</a></li>
    <li><a href="https://gist.github.com/bdaac9f300b65a016cb6533a55b4ce28">python39Packages.qiskit-terra</a></li>
  </ul>
</details>
<details>
  <summary>9 packages skipped due to time constraints:</summary>
  <ul>
    <li>chia</li>
    <li>python38Packages.johnnycanencrypt</li>
    <li>python38Packages.qiskit</li>
    <li>python38Packages.qiskit-aqua</li>
    <li>python38Packages.tumpa</li>
    <li>python39Packages.adblock</li>
    <li>python39Packages.qiskit</li>
    <li>python39Packages.qiskit-aqua</li>
    <li>python39Packages.qiskit-ignis</li>
  </ul>
</details>
<details>
  <summary>26 packages built successfully:</summary>
  <ul>
    <li>dyndnsc</li>
    <li>maturin</li>
    <li>python38Packages.adblock</li>
    <li>python38Packages.clvm-rs</li>
    <li>python38Packages.dependency-injector</li>
    <li>python38Packages.fastapi</li>
    <li>python38Packages.fastdiff</li>
    <li>python38Packages.graphene</li>
    <li>python38Packages.json-logging</li>
    <li>python38Packages.retworkx</li>
    <li>python38Packages.snapshottest</li>
    <li>python38Packages.starlette</li>
    <li>python38Packages.wasmer</li>
    <li>python39Packages.clvm-rs</li>
    <li>python39Packages.dependency-injector</li>
    <li>python39Packages.fastapi</li>
    <li>python39Packages.fastdiff</li>
    <li>python39Packages.graphene</li>
    <li>python39Packages.johnnycanencrypt</li>
    <li>python39Packages.json-logging</li>
    <li>python39Packages.retworkx</li>
    <li>python39Packages.snapshottest</li>
    <li>python39Packages.starlette</li>
    <li>python39Packages.tumpa</li>
    <li>python39Packages.wasmer</li>
    <li>qutebrowser</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/development/tools/rust/maturin/default.nix [hunk 1]
```diff
@@ -5,26 +5,27 @@
...
   };
 
-  cargoSha256 = "113i7a5hpz7qch45wwapp53ixvgssnlxm0810yjicxp3k61ak30j";
+  cargoHash = "sha256-xLT3QeLDZnPrW1rCa0E1DrCYiqhb04o3vdi8pdJdWJQ=";
 
   nativeBuildInputs = [ pkg-config ];
 
   buildInputs = lib.optional stdenv.isLinux dbus
-    ++ lib.optional stdenv.isDarwin Security;
+    ++ lib.optional stdenv.isDarwin [ Security libiconv ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/rust/maturin/default.nix:28]:
```suggestion
    ++ lib.optionals stdenv.isDarwin [ Security libiconv ];
```

Comment (danieldk) [reply pkgs/development/tools/rust/maturin/default.nix]: Whoops, fixed.

**Comment by danieldk** [comment]
Upstream has release another version, so I have updated this PR to 0.10.6.

**Comment by danieldk** [comment]
Merging, since dependencies build correctly and all comments have been adressed.

**Pull request merged**